core/types.ml

```diff
@@ -1995,28 +2009,197 @@ module Policy = struct
                 |> sync)
 
   let effect_sugar
-  = Settings.(flag "effect_sugar"
-              |> synopsis "Toggles the effect sugar in pretty printer."
-              |> convert parse_bool
-              |> sync)
+    = Settings.(flag "effect_sugar"
+                |> synopsis "Toggles the effect sugar in pretty printer."
+                |> convert parse_bool
+                |> sync)
 
-  type t = { quantifiers  : bool
-           ; flavours     : bool
-           ; hide_fresh   : bool
-           ; kinds        : kind_policy
-           ; effect_sugar : bool }
+  module EffectSugar : sig
+    type opt = PresenceOmit
+             | AliasOmit
+             (* vvv TODO name; for now it's the One Effect To Rule Them All *)
+             | ArrowsShowTheOneEffect
+             | ArrowsCurriedCollectionAssumeFresh
+             | ContractOperationArrows
+             | OpenDefault
+             (* | DifferentOperationArrows *)
+             | FinalArrowSharesWithAlias
```

Comment:
This is handled by #1032. In your example, this setting would cause the type to desugar into
```links
f = fun : (Comp(a, { |e})) -e-> Comp(a, { |e})
                      ^-----^--------------^
```

The printer will handle each case correctly. I've included the setting in this PR so that it's compatible with #1032.

Reply:
Hmm. This is not great. It is problematic that the arrow gets endowed with e.  I think we may need to think more carefully about this case.

Reply:
I can disable this setting in this PR and then just reenable it in the other one.
The idea behind this is
```links
sig upd_map : (a, b, Map(a,b)) ~> Map(a,b)
# which would mean
sig upd_map : (a, b, Map(a,b,{ |e})) ~e~> Map(a,b,{ |e})
```
where `Map` contains a function that is relevant to its transformation, and so its effect has to be shared with the arrows as well.

Reply:
Hmm. We need to be careful here. It is often incorrect to assume that the last arrow and the alias should share the effect variable, though, sometimes it is correct. We need to think more carefully about a robust design.